src/combinatorics/rado/finset_section.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+
+import combinatorics.rado.auxiliary
+
+/-!
+# Sections of families of finite sets
+
+Let `F : ι → finset α` be a family of finite sets. A *section* of `F` is a function `f : ι → α`
+such that `f i ∈ F i` for all `i : ι`. We define `finset.is_section F f` to express this property
+and `finset.is_section_on F f s` to express that `f` is a section on the finset `s`.
+
+We then provide some basic API lemmas.
+
+(Note: `section` is a reserved word, so would be problematic as a name.)
+-/
+
+namespace finset
+
+variables {ι : Type*} {α : Type*}
+
+/-- A function `f : ι → α` is a *section* of `F : ι → finset α` if it maps `i` into `F i`. -/
+def is_section (F : ι → finset α) (f : ι → α) : Prop := ∀ i, f i ∈ F i
+
+/-- A function `f : ι → α` is a *section* of `F : ι → finset α` on a finite set `s` if it maps
+`i` into `F i` for all `i ∈ s`. -/
+def is_section_on (F : ι → finset α) (f : ι → α) (s : finset ι) : Prop := ∀ i ∈ s, f i ∈ F i
```

Comment:
Please use semi-implicit arguments on `i`

Reply:
You mean `∀ {{i}} (hi : i ∈ s), ...` ?

Reply:
OK.